src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -325,7 +326,11 @@ final case class EServicesApiServiceImpl(
 
   private def getLatestAgreement(requesterId: UUID, eService: CatalogProcess.EService)(implicit
     contexts: Seq[(String, String)]
-  ): Future[Option[AgreementProcess.Agreement]] =
+  ): Future[Option[AgreementProcess.Agreement]] = {
+
+    val ordering: Ordering[(Int, OffsetDateTime)] =
+      Ordering.Tuple2(Ordering.Int.reverse, Ordering.by(_.toEpochSecond))
```

Comment:
```suggestion
      Ordering.Tuple2(Ordering.Int.reverse, Ordering.by(_.toEpochSecond).reverse)
```
as we need the latest agreement